### ISSUE TITLE
Introductie XML schema constructies voor minimumwaarde en maximumwaarde

### ISSUE THREAD

**Issue by ArjanLoeffen** [Imvertor/Imvertor-Maven#173]
Tagged values minimumwaarde (CFG-TV-MINIMUMVALUE) en maximumwaarde (CFG-TV-MAXIMUMVALUE) komen niet voor in de schema generator voor BRO. Het zijn dus puur informatieve tagged values. 
Dit met nu ook in de schema's terugkeren (als minInclusive en maxInclusive).

Dus dan wil je en willekeurige waarde kunnen opgeven als minimum of als maximum, dus er is dan een "schaal" waarop je die waarden kunt vergelijken (integers, datums, strings of whatever). In XML schema wereld zou dat dan wel expliciet zijn, gezien de "ordered"property: https://www.w3.org/TR/xmlschema-2/#dt-ordered

NB Op dit moment worden minInclusive en maxInclusive gegenereerd in de schema's op basis van de waarde Lengte (CFG-TV-LENGTH), zie ook GIThub #159.
Een lengte 2..5 op een integer waarde wordt omgezet in 

```
<xs:restriction base="xs:integer">
         <xs:minInclusive value="10"/>
         <xs:maxInclusive value="99999"/>
</xs:restriction>
```

**Comment by wilkoquak** [comment]
Ik zou inderdaad graag iets meer controle willen hebben over het xml schema wat gegenererd wordt. In het hierbovenstaande voorbeeld worden minInclusive berekend op basis van het mogelijke aantal digits van het getal dat in de tagged value LENGTH wordt opgeslagen. Voor IMGeluid is meer controle nodig over de gegenereerde facets.

Voorbeelden: 

-  Absorptiefactie is een waarde tussen 0 en 1 met twee cijfers achter de komma (minInclusive="0" maxInclusive="1" fractionDgits="2"
- Amplitude tussen -99.9 en 99.9 (minExclusive="-100" maxExclusive="100" fractiondigits="1").

Oplossingsvoorstel: Een directere mapping tussen de tagged values in het UML en de facets in het xsd.

**Comment by kad-mesdat** [comment]
Is dat dan niet iets voor een reguliere expressie?

**Comment by adbgnm** [comment]
In de XSD's van LV-BAG zie ik inderdaad een pattern staan. Zijn deze xsd's via Imvertor gegenereerd?

**Comment by kad-mesdat** [comment]
Ja, je kunt een TV Formeel patroon toevoegen en dan wordt dat als restriction pattern opgenomen in de xsd.

**Comment by wilkoquak** [comment]
De reguliere expressies voor xs:decimal voor bovengenoemde voorbeelden zijn niet echt makkelijk. Ze hebben die facets in xml schema niet voor niets bedacht.

**Comment by wilkoquak** [comment]
Het is alweer een tijdje stil aangaande dit issue. Ik heb nog steeds de behoefte om xsd te generen met een xs:decimal erin waar middels fractionDigits het maximum aantal cijfers achter de komma is vastgelegd. Hoe krijg is dat met Imvertor voor elkaar?

**Comment by ArjanLoeffen** [comment]
Voor het aantal cijfers voor en achter de komma kun je
> 2.8.2.23 Metagegeven: Lengte (domein van een waarde van een gegeven)
volgen. Dat is geimplementeerd.

Voor de minimum waarde en maximum waarde zou je behoefte hebben aan (bijvoorbeeld) een tagged value "minimum waarde" en "maximum waarde". Die zijn wel voor de metamodellen van VNG gedefinieerd (CFG-TV-MINVALUEINCLUSIVE en CFG-TV-MAXVALUEINCLUSIVE) maar niet voor Geonovum. We kunnen die toevoegen. Ze zijn nu beschikbaar op attributen en data elementen. Is dat wat je zoekt?

**Comment by ArjanLoeffen** [comment]
Ik heb een implementatie gemaakt voor Geonovum in de ISO19136 schema's. 
Voorbeeld:

![image](https://user-images.githubusercontent.com/12828217/118277951-ac797800-b4c9-11eb-8ddf-0a1007267777.png)

met tagged values 

![image](https://user-images.githubusercontent.com/12828217/118278011-c3b86580-b4c9-11eb-91c5-ac8e1d56c236.png)

levert dan op:

![image](https://user-images.githubusercontent.com/12828217/118277909-a08db600-b4c9-11eb-8637-bc6dc29e45fc.png)
 
het betreft dus alleen "inclusive" grenzen. Als meer nodig is (kan relevant zijn bij gebroken getallen) kan dat worden toegevoegd. Let op, de tagged values kunnen willekeurige waarden hebben (dus ook datums en strings) maar alleen numerieke waarden keren zo terug in de schema's. In de documentatie komt wel de min/max waarde terug.

**Comment by wilkoquak** [comment]
Dank, ik heb er nog even over doorgedacht en gekeken naar de use-case in IMGeluid. Het aantal digits voor de komma wordt daar bepaald door Maximumwaarde en Minimumwaarde. Ik zou die niet nog een keer willen specificeren. Het aantal cijfers achter de komma (fractiondigits) is wel van belang. De tekst in het MIM is niet helemaal waterdicht maar als ik met tagged value "Lengte=',2'" kan aageven dat het aantal digits voor de komma met niet uitmaakt en ik alleen het aantal achter de komma wil specificeren ben ik waar ik wezen wil.

**Comment by ArjanLoeffen** [comment]
Dat kunnen we wel zo bouwen maar dat is een aanpassing op MIM11. Wil je dat in gang zetten?

Ik zou dan kiezen voor: Lengte = *,2

dat is iets explicieter.

**Comment by wilkoquak** [comment]
Gedaan! Zie https://github.com/Geonovum/MIM-Werkomgeving/issues/153

**Comment by wilkoquak** [comment]
Deze feature is gebouwd en hij werkt en ik ben er helemaal blij mee. Er zit echter nog 1 bugje in: Bij een "Lengte=*,0" verwacht ik een fractionDigits=0 facet in het schema. In dit geval wordt er echter helemaal geen fractionDigits facet gegenereerd. Bij andere waardes voor Lengte gaat het wel goed.

**Comment by wilkoquak** [comment]
Kan jij deze sluiten Arjan
